vault/data_source_kubernetes_auth_backend_role_test.go

```diff
@@ -157,28 +165,30 @@ func TestAccKubernetesAuthBackendRoleDataSource_full(t *testing.T) {
 						"token_num_uses", "0"),
 					resource.TestCheckResourceAttr("data.vault_kubernetes_auth_backend_role.role",
 						"token_period", "900"),
+					resource.TestCheckResourceAttr("data.vault_kubernetes_auth_backend_role.role",
+						"audience", audience),
```

Comment:
To ensure backwards compatibility, would it be possible to also add a test _without_ the audience field and make sure it doesn't result in unnecessary schema diffs?

Reply:
Sorry about that! I just realized that I've added the audience to the basic and full role configs!
Since audience is not mandatory param it should not be in basic role config which will address your concern.  PTAL https://github.com/terraform-providers/terraform-provider-vault/pull/601/commits/e3e4bcbaba774a5dfe2abfdb13428072e13d48cb